ui/src/app/settings/views/Repositories/RepositoryForm/RepositoryForm.js

```diff
@@ -50,34 +70,101 @@ export const RepositoryForm = ({ title }) => {
+      distributions: repository.distributions.join(", "),
+      enabled: repository.enabled,
+      key: repository.key,
+      name: repository.name,
+      url: repository.url
+    };
+  } else {
+    title = `Add ${typeString}`;
+    initialValues = {
+      arches: ["i386", "amd64"],
+      components: "",
+      default: false,
+      disable_sources: false,
+      disabled_components: [],
+      disabled_pockets: [],
+      distributions: "",
+      enabled: true,
+      key: "",
+      name: "",
+      url: type === "ppa" ? "ppa:" : ""
+    };
+  }
+
   return (
     <>
       {!allLoaded ? (
         <Loader text="Loading..." />
       ) : (
         <Card highlighted className="repo-form">
           <Row>
-            <Col size="3">
+            <Col size="2">
               <h4>{title}</h4>
             </Col>
-            <Col size="7">
-              <div className="repo-form__buttons">
-                <Button
-                  appearance="base"
-                  className="u-no-margin--bottom"
-                  onClick={() => window.history.back()}
-                  type="button"
-                >
-                  Cancel
-                </Button>
-                <Button
-                  appearance="positive"
-                  className="u-no-margin--bottom"
-                  type="button"
-                >
-                  Save repository
-                </Button>
-              </div>
+            <Col size="8">
+              <Formik
+                initialValues={initialValues}
+                validationSchema={RepositorySchema}
+                onSubmit={(values, { resetForm }) => {
+                  const payload = {
+                    arches: values.arches,
+                    components: values.components.split(" ,").filter(Boolean),
+                    default: values.default,
+                    disable_sources: values.disable_sources,
+                    disabled_components: values.disabled_components,
+                    disabled_pockets: values.disabled_pockets,
+                    distributions: values.distributions
+                      .split(" ,")
+                      .filter(Boolean),
+                    enabled: values.enabled,
+                    key: values.key,
+                    name: values.name,
+                    url: values.url
+                  };
+                  console.log(payload);
+                  resetForm(values);
+                }}
+                render={formikProps => (
+                  <Form onSubmit={formikProps.handleSubmit}>
+                    <RepositoryFormFields
+                      formikProps={formikProps}
+                      type={type}
+                    />
+                    <div className="repo-form__buttons">
+                      <Button
+                        appearance="base"
+                        className="u-no-margin--bottom"
+                        onClick={() => window.history.back()}
```

Comment:
90% sure that you don't want to use `window.history` directly and instead should use the [react-router history api](https://reacttraining.com/react-router/web/api/history), but as there's currently no hooks support, perhaps we want to give [use-react-router](https://github.com/CharlesStover/use-react-router) a go?

Reply:
@squidsoup Yeah it felt wrong. Anyway we already had a custom `useRouter` hook in `base/hooks.js` that uses the same context as `use-react-router` does under the hood, so I've used that. There's already a comment to replace them when official `react-router` hooks are available.